packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -433,33 +433,66 @@ class ProductVariantInventoryService extends TransactionBaseService {
         }
       )
 
-    if (reservationCount) {
-      let reservation = reservations[0]
+    reservations.sort((a, _) => {
```

Comment:
suggestion(non blocking): Could this method take an argument like 
```ts
data: {
  lineItemId: string,
  variantId: string,
  locationId: string,
  quantity: number
}[]
```

And make this method able to manage bulk operations? That way we can mutualise the queries to fetch the variants, reservations etc and reduce the overhead in the event loop and the db when wanting to adjust multiple items. I am thinking of those carts that get bigger or if someone use the module as a service in his arch. wdyt?

It might be worth doing that kind of implementation from the beginning to avoid having to do it later. Also, some other methods in this service could do the same. Also another point but not linked to this immediately, but when using `save` to create/update a tuple, typeorm does a select to check if it exists or not before doing the query. I don't think we can of the subscribers/listeners in the module, so for the case of a create since we are already doing a find ourselves, I would suggest to do an insert or insertBulk (see image, money amount repos). Same for the update, a save will do a read before updating but we already do the read. wdyt?

cc @srindom

Reply:
Great idea - although wouldn't be a requirement from my end at this point

Reply:
Keeping this as is for now, but I generally agree that we should try to default to bulk operations where possible 😄